concert/app/test/aggregate_writer/test_aggregate_writer.py

```diff
@@ -0,0 +1,151 @@
+                                "CHOICE_COUNT": choice_count,
+                                "CHOICE_TYPE": choice_type,
+                                "CHOICE_TIME": test_timestamp,
+                            }
+                        )
+                    )
+                ),
+            }
+        ]
+    }
+
+
+@pytest.mark.parametrize(
+    "choice_type, choice_sum, choice_count, choice_key, emotion_or_color",
+    [
+        ("CHOICE_COLOR_#00AB00", 10.0, 5, "colors", "#00AB00"),
+        ("CHOICE_EMOTION_HAPPINESS", 12.0, 7, "emotions", "EMOTION_HAPPINESS"),
+        ("CHOICE_EMOTION_ENERGY", 0.5, 1, "emotions", "EMOTION_ENERGY"),
+        ("CHOICE_CHILLS", 1.5, 3, "chills", None),
+    ],
+)
+def test_choices(choice_type, choice_sum, choice_count, choice_key, emotion_or_color):
+    event = get_event(
+        choice_sum=choice_sum, choice_count=choice_count, choice_type=choice_type
+    )
+
+    expected_map_creation_params = {
+        "TableName": table_name,
+        "Key": {"recordId": "AGGREGATE"},
+        "UpdateExpression": "SET #choice_key.#timestamp = :empty_map",
+        "ExpressionAttributeNames": {
+            "#choice_key": choice_key,
+            "#timestamp": test_timestamp,
+        },
+        "ExpressionAttributeValues": {":empty_map": {}},
+        "ConditionExpression": "attribute_not_exists(#choice_key.#timestamp)",
+        "ReturnValues": "NONE",
+    }
+
+    expected_aggregate_update_params = {
+        "TableName": table_name,
+        "Key": {"recordId": "AGGREGATE"},
+        "UpdateExpression": "ADD #choice_key.#timestamp.#choice_string :choice_sum, #choice_key.#timestamp.#total_choices :choice_count",
+        "ExpressionAttributeNames": {
+            "#choice_key": choice_key,
+            "#timestamp": test_timestamp,
+            "#total_choices": "choices",
+        },
+        "ExpressionAttributeValues": {
+            ":choice_sum": Decimal(choice_sum),
+            ":choice_count": choice_count,
+        },
+        "ReturnValues": "NONE",
+    }
+
+    if choice_type.startswith("CHOICE_COLOR"):
+        expected_aggregate_update_params["ExpressionAttributeNames"][
+            "#choice_string"
+        ] = emotion_or_color
+
+    elif choice_type.startswith("CHOICE_EMOTION"):
+        expected_aggregate_update_params["ExpressionAttributeNames"][
+            "#choice_string"
+        ] = emotion_or_color
+        expected_aggregate_update_params["ExpressionAttributeNames"][
+            "#total_choices"
+        ] = "choices_{}".format(emotion_or_color)
+
+    elif choice_type == "CHOICE_CHILLS":
+        expected_aggregate_update_params["ExpressionAttributeNames"][
+            "#choice_string"
+        ] = "chills"
+
+    resp = None
+    with Stubber(dynamodb.meta.client) as stub:
+        stub.add_response(
+            "update_item",
+            dynamo_update_item_success_response,
+            expected_map_creation_params,
+        )
+        stub.add_response(
```

Comment:
You shouldn't need to test every service call is each test. Each test should test a single thing.

Reply:
Fair ask. I've tried to do a better job of this in the most recent revision.